cw_monero/lib/api/transaction_history.dart

```diff
@@ -13,15 +14,23 @@ import 'package:monero/src/generated_bindings_monero.g.dart' as monero_gen;
 import 'package:mutex/mutex.dart';
 
 
+Map<int, Map<String, String>> txKeys = {};
 String getTxKey(String txId) {
+  txKeys[wptr!.address] ??= {};
+  if (txKeys[wptr!.address]![txId] != null) {
+    return txKeys[wptr!.address]![txId]!;
+  }
   final txKey = monero.Wallet_getTxKey(wptr!, txid: txId);
   final status = monero.Wallet_status(wptr!);
   if (status != 0) {
-    final error = monero.Wallet_errorString(wptr!);
+    monero.Wallet_errorString(wptr!);
+    txKeys[wptr!.address]![txId] = "";
```

Comment:
shouldn't we try again the next time?
like this if this errors out, it will cache it as empty string, thus it will be always empty, while it should try again every time

Reply:
If we don't have the key it won't change in the future, this is to save times and not log too much to console.

Reply:
We call these functions to not leak errors in other places